rllib/offline/dataset_reader.py

```diff
@@ -113,23 +115,30 @@ def __init__(self, ioctx: IOContext, ds: ray.data.Dataset):
         """
         self._ioctx = ioctx
         self._dataset = ds
+        # the number of rows to return per call to next()
+        batch_size = ioctx.config.get("train_batch_size", 1)
+        num_workers = ioctx.config.get("num_workers", 0)
+        if num_workers:
+            batch_size = max(math.ceil(batch_size / num_workers), 1)
         # We allow the creation of a non-functioning None DatasetReader.
         # It's useful for example for a non-rollout local worker.
         if ds:
             print(
                 "DatasetReader ", ioctx.worker_index, " has ", ds.count(), " samples."
             )
-            self._iter = self._dataset.repeat().iter_rows()
+            self._iter = self._dataset.repeat().iter_batches(batch_size=batch_size)
```

Comment:
if you are creating a for loop in next() anyways, I don't think you need to compute the batch_size and do iter_batches() here?
we should simply keep self._iter as iter_rows(). and maybe save batch_size as self._batch_size in the __init__(). and then going through 1 row/batche at a time in next() until you have collected enough samples (>= self._batch_size).

the reason being, we don't actually know the format of data specified here. each row/batch of data may contain a lot of (or variable num of) samples. so we can't simply hardcode here to say that you want a batch of 1024, and you have 16 workers, so I am gonna ask dataset for 64 rows every time. Since those 64 rows may contain tons of data.

makes sense?